code/tests/_game_test.dm

```diff
@@ -94,3 +94,67 @@
 		instance = new type()
 	LAZYADD(allocated, instance)
 	return instance
+
+/datum/game_test/proc/puppeteer(carbon_type = /mob/living/carbon/human, turf/initial_location)
+	TEST_ASSERT(ispath(carbon_type, /mob/living/carbon/human), "unexpected puppeteer carbon type [carbon_type]")
+	var/datum/test_puppeteer/puppeteer = new
+
+	if(!initial_location)
+		initial_location = locate(179, 136, 1) // Center of admin testing area
+	puppeteer.origin_test = src
+	puppeteer.puppet = allocate(carbon_type, initial_location)
+	var/datum/mind/new_mind = new("interaction_test_[puppeteer.puppet.UID()]")
+	new_mind.transfer_to(puppeteer.puppet)
+
+	return puppeteer
+
+/datum/test_puppeteer
```

Comment:
I would prefer if this datum was in its own file, but considering how unit tests has their own special handling of files, if you dont want to do it its fine with me

Reply:
That's no prob, I can pull it into its own file